airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -576,19 +578,21 @@ def load_file_obj(self,
             object to be uploaded
         :type acl_policy: str
         """
-        self._upload_file_obj(file_obj, key, bucket_name, replace, encrypt, acl_policy)
+        self._upload_file_obj(file_obj, key, bucket_name, replace, encrypt, acl_policy, extra_args)
 
     def _upload_file_obj(self,
                          file_obj,
                          key,
                          bucket_name=None,
                          replace=False,
                          encrypt=False,
-                         acl_policy=None):
+                         acl_policy=None,
+                         extra_args=None):
         if not replace and self.check_for_key(key, bucket_name):
             raise ValueError("The key {key} already exists.".format(key=key))
 
-        extra_args = {}
+        if not extra_args:
```

Comment:
You should make a deepcopy of the extra_args otherwise you can mutate external args with the next line.

Reply:
Hi, thanks for the feedback! Are you talking about something like this?
```        
if not extra_args:
            extra_args = copy.deepcopy(extra_args)
            extra_args = {}
```

Im not sure I understand which external args you are referring to.

Reply:
@potiuk